### ISSUE TITLE
Context based Logging System

### ISSUE THREAD

**Issue by gingerBill** [odin-lang/Odin#240]
Add logging system to the context

General idea (not proposal)

```
package log 

Logger_Mode :: struct {
    None,
    Minimal,
    General,
    Verbose,
}

Logger_Proc :: #type proc(
	logger_data: rawptr, mode: Logger_Mode,
	ident: string, text: string, level: int, location := #caller_location,
);

Logger :: struct {
    procedure: Logger_Proc,
    data:      rawptr,
}
```
```
Context :: struct {
	allocator:  mem.Allocator,
	logger:     log.Logger, 
	thread_id:  int,

	user_data:  any,
	user_index: int,

	parent:     ^Context,
	derived:    any, // May be used for derived data types
}
```

**Comment by ThisDrunkDane** [comment]
Hasn't this been implemented or has it just been worked on a little @gingerBill?

**Comment by gingerBill** [comment]
This has not been implemented yet as I do not know the best "format" for the logger. Odin does have the `assertion_failure_proc` which is similar but for assertion-like calls.

**Comment by ThisDrunkDane** [comment]
Well for the different "logger levels/mode" at work we use log4net (which is a port of an apache thing). But it has `Debug, Info, Warning, Error, Fatal` and that has served us really well, they then have a xml config file to define what levels to print, that I could live without! But that would be up to the different logger implementations on how to handle that

**Comment by ThisDrunkDane** [comment]
This was implemented by PR #276 a while ago, so I'm closing this
